### PR TITLE
Adding json-output for api-like functionality

### PR DESCRIPTION
**Pull request by mr-deamon**

### CODE REVIEW

File: snappass/main.py [hunk 1]
```diff
@@ -162,6 +162,7 @@ def index():
 @app.route('/', methods=['POST'])
 def handle_password():
     ttl, password = clean_input()
+    json = request.form.get('json',False,type=bool)
```
Comment (jparise) [inline snappass/main.py:165]:
```suggestion
    json = request.form.get('json', False, type=bool)
```

**Review by jparise — COMMENTED** [review]
Looks good! Could you add some test coverage for this change, too?

File: snappass/main.py (hunk 1 above)
Comment (devinlundberg) [reply snappass/main.py:165]: Its probably better to switch on the accept header which flask already parses for us: https://flask.palletsprojects.com/en/1.0.x/api/#flask.Request.accept_mimetypes

File: snappass/main.py [hunk 2]
```diff
@@ -177,7 +178,10 @@ def handle_password():
     if URL_PREFIX:
         base_url = base_url + URL_PREFIX.strip("/") + "/"
     link = base_url + url_quote_plus(token)
-    return render_template('confirm.html', password_link=link)
+    if json:
+        return render_template('confirm.json', password_link=link, ttl=ttl)
```
Comment (jparise) [inline snappass/main.py:182]: We could use `flask.jsonify` here to also set the appropriate response headers:

```py
return jsonify(link=link, ttl=ttl)
```

**Comment by mr-deamon** [comment]
What about something like this? Sorry, tests are not really my strongest topic :)

File: snappass/templates/confirm.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"snappass":{"link":"{{ password_link }}","ttl":"{{ ttl }}"}}
```
Comment (jparise) [inline snappass/templates/confirm.json:1]: You no longer need this template.

File: snappass/main.py (hunk 1 above)
Comment (jparise) [reply snappass/main.py:165]: Per the suggestion, you can drop this line and below do:

```py
if request.accept_mimetypes.accept_json:
    return jsonify(link=link, ttl=ttl)

return render_template('confirm.html', password_link=link)
```

File: tests.py [hunk 4]
```diff
@@ -138,6 +139,21 @@ def test_set_password(self):
 
             frozen_time.move_to("2020-05-22 12:00:00")
             self.assertIsNone(snappass.get_password(key))
+    
+    def test_set_password_json(self):
+        with freeze_time("2020-05-08 12:00:00") as frozen_time:
+            password = 'my name is my passport. verify me.'
+            rv = self.app.post('/', data={'password': password, 'ttl': 'two weeks', 'json': True})
+
+            json_content = rv.data.decode("ascii")
```
Comment (jparise) [inline tests.py:148]: JSON should be treated as UTF-8, but that aside, there's a simple approach you can use here:

```suggestion
            json_content = rv.get_json()
```

... and then you don't need to use `json.loads()` below.

File: snappass/templates/confirm.json (hunk 3 above)
Comment (mr-deamon) [reply snappass/templates/confirm.json:1]: yep, forgot to commit, sorry!

**Review by jparise — APPROVED** [review]

**Comment by jparise** [comment]
Thanks for the contribution!

**Pull request merged**